src/validator.ts

```diff
@@ -188,49 +189,268 @@ export const possiblyReturningBlock = warning<Method>(node => {
+  (node.name ?? '').replaceAll('"', '').trim() !== ''
+)
+
+export const shouldNotMarkMoreThanOneOnlyTest = warning<Test>(node =>
+  !node.isOnly || node.parent().children().filter(element => element.is('Test') && element.isOnly).length <= 1
+)
+
+export const shouldNotDefineNativeMethodsOnUnnamedSingleton = error<Method>(node => {
+  const parent = node.parent()
+  return !node.isNative() || !parent.is('Singleton') || !!parent.name
+})
+
 // ══════════════════════════════════════════════════════════════════════════════════════════════════════════════════
 // HELPER FUNCTIONS
 // ══════════════════════════════════════════════════════════════════════════════════════════════════════════════════
 
 const allParents = (module: Module) =>
   module.supertypes.map(supertype => supertype.reference.target()).flatMap(supertype => supertype?.hierarchy() ?? [])
 
+const getReferencedModule = (parent: Node): Module | undefined => {
+  if (parent.kind === 'ParameterizedType') return parent.reference.target()
+  if (parent.kind === 'New') return parent.instantiated.target()
+  return undefined
+}
+
+const uninitializedValue = (value: Expression | undefined) => value && value.is('Literal') && !value.value && value.isSynthetic()
+
+const getUninitializedAttributes = (node: Module, initializers: string[] = []): string[] => {
+  const uninitializedAttributes: string[] = []
+  node.defaultFieldValues()?.forEach(
+    (value, field) => {
+      if (uninitializedValue(value) && !initializers.includes(field.name)) {
+        uninitializedAttributes.push(field.name)
+      }
+    })
+  return uninitializedAttributes
+}
+
+const isBooleanLiteral = (node: Expression, value: boolean) => node.is('Literal') && node.value === value
+
+const targetSupertypes = (node: Class | Singleton) => node.supertypes.map(_ => _?.reference.target())
+
+const superclassMethod = (node: Method) => node.parent().lookupMethod(node.name, node.parameters.length, { lookupStartFQN: node.parent().fullyQualifiedName(), allowAbstractMethods: true })
+
+const finishesFlow = (sentence: Sentence, node: Node): boolean => {
+  const parent = node.parent()
+  const lastLineOnMethod = parent.is('Body') ? last(parent.sentences) : undefined
+  const returnCondition = (sentence.is('Return') && lastLineOnMethod !== node && lastLineOnMethod?.is('Return') || lastLineOnMethod?.is('Throw')) ?? false
+  return sentence.is('Throw') || sentence.is('Send') || sentence.is('Assignment') || sentence.is('If') || returnCondition
+}
+
+const isGlobal = (node: Variable) => node.parent().is('Package')
+
+const getVariableContainer = (node: Node): Method | Test => {
+  let nodeContainer = node.parent()
+  while (!nodeContainer.is('Method') && !nodeContainer.is('Test')) {
+    nodeContainer = nodeContainer.parent()
+  }
+  return nodeContainer
+}
+
+const getAllReferences = (node: Method | Test): List<Variable> => node.sentences().filter(sentence => sentence.is('Variable')) as List<Variable>
```

Comment:
Me parece medio confuso que el método se llame `getAllReferences` pero traiga `Variables`
Por otro lado, si usas la función `is` en lugar del mensaje is te ahorras el casteo:

`node.sentences().filter(is('Variable'))`

Reply:
Ok, puede ser que resulte confuso cuando estás haciendo el motor del intérprete. Vamos a ponerle variables para que no haya problemas (yo lo usaba en el contexto de OO y me sonaba totalmente ok).